tox.ini

```diff
@@ -58,6 +59,7 @@ install_command = python -m pip install --no-deps {packages}
 commands_pre =
     # Generate a requirements-min.txt file
     minimum_deps
+    python -m pip install --upgrade pip
```

Comment:
So in the CI workflow we have a step that upgrades `pip` automatically.  So this is redundant.  It could be argued that all this should be in `tox` instead of `ci.yml`, but then we should do it for all envs, not just a few.